### ISSUE TITLE
Can't delete an email and N1 is not throwing any error.

### ISSUE THREAD

**Issue by bashok001** [nylas/N1#1799]
Can't delete this email from my Inbox. Task is shown as success in Developer console under ChangeLabelsTask. But Delta Streaming shows this message in response to modify thread request.

> "ignoredBecause":"This model is locked by the optimistic change tracker"

I am not sure what is it is but this is what is happening. Even opening the email and clicking delete would not delete it.

![try3](https://cloud.githubusercontent.com/assets/2344626/13910426/64a4f406-eef8-11e5-9a3d-e2cd48d8ea6f.gif)

##### Are there any related issues?

No
##### What operating system are you using?
Arch Linux

##### What version of N1 are you using?
0.4.10

--

**Bug?**
##### Do you have any third-party plugins installed?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Not sure. Currently this is in Gmail

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Provided above.

**Comment by mbilker** [comment]
Version 0.4.16 is out. I believe there have been changes to the `DatabaseStore` to fix issues like this.

**Comment by jstejada** [comment]
hey @bashok001, I think this might be the same as your previous issue #1523 and potentially #1720 .
If you refresh the view, does the message get correctly removed from the view?

**Comment by bashok001** [comment]
Hey @jstejada : I thought so too... Nope, the message stayed on. When I rebooted my machine, the next day, the message was still there (This being 6 hours after this report, at the least). But when I clicked delete this time, it went off. I just kept getting this in the console everytime I tried to delete before reboot. 
> "ignoredBecause":"This model is locked by the optimistic change tracker"

Do note, it may not have needed full reboot. I just didnt quit N1. Also note, only this email had that problem. The ones that arrived after that got deleted properly. If you do think this is related, feel free to close.
